backend/users/autocomplete.py

```diff
@@ -0,0 +1,69 @@
+import json
+
+from django.contrib.admin.widgets import SELECT2_TRANSLATIONS, AutocompleteSelect
+from django.urls import reverse
+from django.utils.translation import get_language
+
+from users.client import get_users_data_by_ids
+
+
+class UsersBackendAutocomplete(AutocompleteSelect):
+    def __init__(self, admin_site, attrs=None, choices=(), using=None):
+        self.admin_site = admin_site
+        self.db = using
+        self.choices = choices
+        self.attrs = {} if attrs is None else attrs.copy()
+        self.i18n_name = SELECT2_TRANSLATIONS.get(get_language())
+
+    def build_attrs(self, base_attrs, extra_attrs=None):
+        """
+        Set select2's AJAX attributes.
+        Attributes can be set using the html5 data attribute.
+        Nested attributes require a double dash as per
+        https://select2.org/configuration/data-attributes#nested-subkey-options
+        """
+        attrs = {}
+        attrs.setdefault("class", "")
+        attrs.update(
+            {
+                "data-ajax--cache": "true",
+                "data-ajax--delay": 250,
+                "data-ajax--type": "GET",
+                "data-ajax--url": reverse("admin:users-admin-autocomplete"),
+                "data-theme": "admin-autocomplete",
+                "data-allow-clear": json.dumps(not self.is_required),
+                "data-placeholder": "",  # Allows clearing of the input.
+                "lang": self.i18n_name,
+                "class": attrs["class"]
+                + (" " if attrs["class"] else "")
+                + "admin-autocomplete",
+            }
+        )
+        return attrs
```

Comment:
same as above

we could also move these to a base class if we are going to reuse them

Reply:
a few things are custom here, what is different is basically no reference to the field and the url is hardcoded there

you can see the original here: https://github.com/django/django/blob/8eb56930915f8022aa27ec29ab5be7d21764608c/django/contrib/admin/widgets.py#L396

Reply:
>we could also move these to a base class if we are going to reuse them

the base class AutocompleteSelect is not something we own but part of django, or are you referring to something else?

Reply:
we can create our own base class :) but only if we use it more than once otherwise it is fine

Reply:
If we add more data from external services it might make sense yes, but ideally I would prefer the users service to be the only external service to call, otherwise we risk building a spider web of services calling each other :'D